File: Chapter7/g.cpp
#include <iostream>

using namespace std;

string spaceRemover(string s) {
    string res = "";
    for (int i = 0; i < s.length(); i++) {
        if (s[i] != ' ' && s[i] != '_') {
            res += s[i];
        }
    }
    return res;
}

string reverseString(string s) {
    string res = "";
    for (int i = 0; i < s.length(); i++) {
        res += s[s.length() - i - 1];
    }
    return res;
}

bool isPalindrome(string s) {
    return s == reverseString(s);
}

int main() {
    string s;
    getline(cin, s);

    if (isPalindrome(spaceRemover(s))) {
        cout << "True\n";
    } else {
        cout << "False\n";
    }
    return 0;
}

File: Chapter7/g.pas
var
    s: String;

function space_remover(s: String): String;
    var 
        res: String;
        i: Integer;
    Begin
        res := '';
        for i:= 1 to length(s) do
            if not((s[i] = ' ') or (s[i] = '_')) then
                res := res + s[i];
        space_remover := res;
    End;

function reverse(s: String): String;
    var 
        res: String;
        i: Integer;
    Begin
        res := '';
        for i:= length(s) downto 1 do
            res := res + s[i];
        reverse := res;
    End;

function is_palindrome(s: String): Boolean;
    Begin
        is_palindrome := reverse(s) = s;
    End;




Begin
    readln(s);
    writeln(is_palindrome(space_remover(s)));
End.
